src/main/drivers/system.c

```diff
@@ -39,16 +39,25 @@ static volatile uint32_t sysTickUptime = 0;
 static volatile uint32_t sysTickValStamp = 0;
 // cached value of RCC->CSR
 uint32_t cachedRccCsrValue;
+static uint32_t cpuClockFrequency = 0;
 
 void cycleCounterInit(void)
 {
 #if defined(USE_HAL_DRIVER)
-    usTicks = HAL_RCC_GetSysClockFreq() / 1000000;
+    cpuClockFrequency = HAL_RCC_GetSysClockFreq();
 #else
     RCC_ClocksTypeDef clocks;
     RCC_GetClocksFreq(&clocks);
-    usTicks = clocks.SYSCLK_Frequency / 1000000;
+    cpuClockFrequency = clocks.SYSCLK_Frequency;
 #endif
+    usTicks = cpuClockFrequency / 1000000;
+
+    CoreDebug->DEMCR |= CoreDebug_DEMCR_TRCENA_Msk;
+#if defined(STM32F7)
+    DWT->LAR = 0xC5ACCE55;
```

Comment:
Should probably use a define instead of a magic number here (albeit this does look 'magic' ;-) ).

Reply:
yeah, it's totally magic, I'll add a reference to the documentation for that too.

Reply:
done, re-tested the CYCCNT register on F303, F405, F722, H750 using this snippet in `main()`

```
        debug[0] = getCycleCounter();
        DWT->CYCCNT = 0;
```
observed that the cycle counter is correctly reset on all MCUs tested.